staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -66,7 +67,10 @@ type ApplyFlags struct {
 	All            bool
 	Overwrite      bool
 	OpenAPIPatch   bool
-	PruneWhitelist []string
+
+	// DEPRECATED: Use PruneAllowlist instead
+	PruneWhitelist []string // TODO: Remove this in kubectl 1.29 or later
```

Comment:
> Assuming kubectl 1.29 is when the old flag can be removed (3 releases per year) if this makes it into 1.26. Is this correct?

This is a good question. Strictly speaking, pruning is in alpha, which requires zero releases (the whitelist flag isn't marked as such, but it requires the --prune flag, which is marked, to do anything). But we've agreed that it needs to be treated at least as beta because of how long it has been around. Per https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli
> CLI elements of user-facing components (e.g. kubectl) must function after their announced deprecation for no less than:
> Beta: 3 months or 1 release (whichever is longer)

I think that means if we announce in 1.26, it needs to work at least in 1.27, but can be removed in 1.28? I'm not against being more generous though.

Reply:
I didn't realize beta features had a different deprecation timeline. I'll update the comment to say 1.28.